R/vaccination.R

```diff
@@ -121,8 +122,555 @@ build_vaccine_progression_rate <- function(vaccine_progression_rate,
+##'  2. all those 80 years of age and over and frontline health and social
+##'     care workers
+##'  3. all those 75 years of age and over
+##'  4. all those 70 years of age and over and clinically extremely
+##'     vulnerable individuals
+##'  5. all those 65 years of age and over
+##'  6. all individuals aged 16 years to 64 years with underlying health
+##'     conditions which put them at higher risk of serious disease and
+##'     mortality
+##'  7. all those 60 years of age and over
+##'  8. all those 55 years of age and over
+##'  9. all those 50 years of age and over
+##' 10. all those 40-49 years of age and over
+##' 11. all those 30-39 years of age and over
+##' 12. all those 18-29 years of age and over
+##' @title Compute vaccination order
+##'
+##' @param uptake A vector of length 19 with fractional uptake per
+##'   group. If a single number is given it is shared across all
+##'   groups (note that this includes under-18s)
+##'
+##' @param prop_hcw Assumed fraction of healthcare workers in each
+##'   group (length 19) - if `NULL` we use a default that is a guess
+##'   with hopefully the right general shape.
+##'
+##' @param prop_very_vulnerable Assumed fraction "very vulnerable" in
+##'   each group (length 19) - if `NULL` we use a default that is a
+##'   guess with hopefully the right general shape.
+##'
+##' @param prop_underlying_condition Assumed fraction "underlying
+##'   condition" in each group (length 19) - if `NULL` we use a
+##'   default that is a guess with hopefully the right general shape.
+##'
+##' @return A matrix with n_groups rows (19) and columns representing
+##'   priority groups, and element (i, j) is the proportion in group i
+##'   who should be vaccinated as part of priority group j, accounting
+##'   for uptake so that the sum over the rows corresponds to the
+##'   total fractional uptake in that group. For
+##'   `vaccine_priority_population`, the total number of
+##'   individuals replaces the proportion (based on the demography
+##'   used by sircovid).
+##'
+##' @rdname vaccine_priority
+##' @export
+vaccine_priority_proportion <- function(uptake,
+                                        prop_hcw = NULL,
+                                        prop_very_vulnerable = NULL,
+                                        prop_underlying_condition = NULL) {
+  n_groups <- carehomes_n_groups()
+
+  if (is.null(uptake)) {
+    uptake <- rep(1, n_groups)
+  } else if (length(uptake) == 1L) {
+    uptake <- rep(uptake, n_groups)
+  } else if (length(uptake) != n_groups) {
+    stop(sprintf("Invalid length %d for 'uptake', must be 1 or %d",
+                 length(uptake), n_groups))
+  }
+
+  prop_hcw <- prop_hcw %||%
+    c(rep(0, 4), rep(0.1, 10), rep(0, 5))
+  prop_very_vulnerable <- prop_very_vulnerable %||%
+    c(rep(0, 4), rep(0.05, 5), rep(0.1, 5), rep(0.15, 5))
+  prop_underlying_condition <- prop_underlying_condition %||%
+    c(rep(0, 4), rep(0.05, 5), rep(0.1, 5), rep(0.15, 5))
```

Comment:
We have very vulnerable and underlying condition proportions for CHR and CHW here - is this necessary given that they are all in the first priority group anyway?

Reply:
they should not be in the same priority group - can you check they are indeed not @edknock ?

Reply:
one is in priority group 4, the other in group 6 
![image](https://user-images.githubusercontent.com/6973431/110362929-d29e1a00-8039-11eb-8871-b964509d5b4d.png)

Reply:
ah sorry I misread your message - I see your question is just for CHR and CHW. Indeed it's not useful but I just think it's easier to manipulate vectors of length 19 all the way through. The values I put for now are totally rubbish anyway, and any proportion > 1 in CHR and CHW should not be used anyway

Reply:
Okay, sounds fine to me then